File: 1.구현/Programmers/[jeunghun] solution.py
def solution(survey, choices):
    scores = {'R': 0, 'T': 0, 'C': 0, 'F': 0, 'J': 0, 'M': 0, 'A': 0, 'N': 0}
    for i, choice in enumerate(choices):
        score = abs(choice - 4)

        if choice > 4:
            scores[survey[i][1]] += score
        else:
            scores[survey[i][0]] += score

    answer = ''
    indexes = ['RT', 'CF', 'JM', 'AN']
    for idx in indexes:
        if scores[idx[0]] < scores[idx[1]]:
            answer += idx[1]
        else:
            answer += idx[0]

    return answer

File: 1.구현/Programmers/(2022 KAKAO TECH INTERNSHIP)성격 유형 검사하기/[jeunghun] solution.swift
import Foundation

func solution(_ survey:[String], _ choices:[Int]) -> String {
    var scores: [Character: Int] = ["R": 0, "T": 0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0]
    
    for (i, choice) in choices.enumerated() {
        let score: Int = abs(choice - 4)
        
        if choices[i] < 4 {
            scores[survey[i].first!]? += score
        } else {
            scores[survey[i].last!]? += score
        }
    }
    
    var answer: String = ""
    let indicators: [String] = ["RT", "CF", "JM", "AN"]
    
    for indicator in indicators {
        let firstChar: Character = indicator.first!
        let lastChar: Character = indicator.last!
        
        if let firstScore = scores[firstChar], let lastScore = scores[lastChar] {
            if firstScore < lastScore {
                answer += String(lastChar)
            } else {
                answer += String(firstChar)
            }
        }
    }
    
    return answer
}